rcl/src/rcl/localhost.c

```diff
@@ -41,9 +41,11 @@ rcl_get_localhost_only(rmw_localhost_only_t * localhost_only)
       get_env_error_str);
     return RCL_RET_ERROR;
   }
-  *localhost_only = (ros_local_host_env_val != NULL &&
-    strcmp(
-      ros_local_host_env_val,
-      "1") == 0) ? RMW_LOCALHOST_ONLY_ENABLED : RMW_LOCALHOST_ONLY_DISABLED;
+  if (ros_local_host_env_val == NULL || ros_local_host_env_val[0] == '\0') {
+    *localhost_only = RMW_LOCALHOST_ONLY_DEFAULT;
+  } else {
+    *localhost_only = strcmp(
+      ros_local_host_env_val, "1") == 0 ? RMW_LOCALHOST_ONLY_ENABLED : RMW_LOCALHOST_ONLY_DISABLED;
```

Comment:
I'm surprised none of the CI has complained about this, but we should use `strncmp` I think, or even just compare the first value of `ros_local_host_env_val` to `'1'`.

Reply:
fixed in 911932f7cbd2741574e520a4f6bc3b81053246f2